packages/rest/src/controllers/didcomm/out-of-band/OutOfBandController.ts

```diff
@@ -0,0 +1,266 @@
+import type { DidCommOutOfBandRecord } from './OutOfBandControllerTypes'
+import type { PlaintextMessage } from '@credo-ts/core/build/types'
+
+import {
+  AgentMessage,
+  JsonTransformer,
+  OutOfBandInvitation,
+  RecordNotFoundError,
+  ConnectionInvitationMessage,
+} from '@credo-ts/core'
+import { parseMessageType, supportsIncomingMessageType } from '@credo-ts/core/build/utils/messageType'
+import { Body, Controller, Delete, Example, Get, Path, Post, Query, Request, Route, Security, Tags } from 'tsoa'
+import { injectable } from 'tsyringe'
+
+import { RequestWithAgent } from '../../../authentication'
+import { apiErrorResponse } from '../../../utils/response'
+import { RecordId } from '../../types'
+import { connectionsRecordExample } from '../connections/ConnectionsControllerExamples'
+import { connectionRecordToApiModel, type DidCommConnectionsRecord } from '../connections/ConnectionsControllerTypes'
+
+import {
+  legacyInvitationExample,
+  outOfBandInvitationExample,
+  outOfBandRecordExample,
+} from './OutOfBandControllerExamples'
+import {
+  outOfBandRecordToApiModel,
+  DidCommOutOfBandCreateInvitationOptions,
+  DidCommOutOfBandCreateLegacyConnectionInvitationOptions,
+  DidCommOutOfBandCreateLegacyConnectionlessInvitationOptions,
+  DidCommOutOfBandReceiveInvitationOptions,
+  DidCommOutOfBandAcceptInvitationOptions,
+} from './OutOfBandControllerTypes'
+
+@Tags('DIDComm Out Of Band')
+@Route('/didcomm/out-of-band')
+@Security('tenants', ['tenant'])
+@injectable()
+export class OutOfBandController extends Controller {
+  /**
+   * Retrieve all out of band records by query
+   */
+  @Example([outOfBandRecordExample])
+  @Get()
+  public async findOutOfBandRecordsByQuery(
+    @Request() request: RequestWithAgent,
+    @Query('invitationId') invitationId?: string,
+  ): Promise<DidCommOutOfBandRecord[]> {
+    const outOfBandRecords = await request.user.agent.oob.findAllByQuery({
+      invitationId,
+    })
+
+    return outOfBandRecords.map(outOfBandRecordToApiModel)
+  }
+
+  /**
+   * Retrieve an out of band record by id
+   */
+  @Example(outOfBandRecordExample)
+  @Get('/:outOfBandId')
+  public async getOutOfBandRecordById(
+    @Request() request: RequestWithAgent,
+    @Path('outOfBandId') outOfBandId: RecordId,
+  ): Promise<DidCommOutOfBandRecord> {
+    const outOfBandRecord = await request.user.agent.oob.findById(outOfBandId)
+
+    if (!outOfBandRecord) {
+      this.setStatus(404)
+      return apiErrorResponse(`Out of band record with id "${outOfBandId}" not found.`)
+    }
+    return outOfBandRecordToApiModel(outOfBandRecord)
+  }
+
+  /**
+   * Creates an outbound out-of-band record containing out-of-band invitation message defined in
```

Comment:
Does TSOA pick this comment up and display it in swagger? Would be cool!

Reply:
Yes it does :)